packages/hebao_v1/contracts/modules/core/ForwarderModule.sol

```diff
@@ -103,27 +101,53 @@ abstract contract ForwarderModule is BaseModule
         );
 
         bytes32 metaTxHash = EIP712.hashPacked(FORWARDER_DOMAIN_SEPARATOR, encoded);
-        require(metaTxHash.verifySignature(from, signature), "INVALID_SIGNATURE");
+
+        // Instead of always taking the expensive path through ER1271,
+        // skip directly to the wallet owner here (which could still be another contract).
+        //require(metaTxHash.verifySignature(from, signature), "INVALID_SIGNATURE");
+        (uint _lock,) = controllerCache.securityStore.getLock(from);
+        require(_lock <= block.timestamp, "WALLET_LOCKED");
+        require(metaTxHash.verifySignature(Wallet(from).owner(), signature), "INVALID_SIGNATURE");
```

Comment:
In ERC1271 module, verify signature is also to check whether it's signed by owner or not. Looks only some extra function call is included. How many gas will it influence with or without the optimization?

Reply:
This saves 11,000 gas. Calls on the wallet are extra expensive because they have to go through the proxy contract, so there are additional CALL and SLOAD costs. There's also some other savings like not having to check twice in the signature code that the signer isn't an address etc...

Reply:
Still 8,000 gas cheaper after #1905.